packages/core/src/render3/component_ref.ts

```diff
@@ -275,6 +276,15 @@ export class ComponentRef<T> extends viewEngine_ComponentRef<T> {
     this.componentType = componentType;
   }
 
+  override setInput(name: string, value: unknown): void {
+    let inputData = this._tNode.inputs;
+    let dataValue: PropertyAliasValue|undefined;
+    if (inputData != null && (dataValue = inputData[name])) {
+      setInputsForProperty(this._rootLView[TVIEW], this._rootLView, dataValue, name, value);
+      markDirtyIfOnPush(this._rootLView, this._tNode.index);
+    }
```

Comment:
Just a couple of small nits:
- can we use `!==` to check input data (would it be `null` if it's not set?)
- we can create a local `lView` var, so that it minifies a bit better (the `_rootLView` part would not be minified) 
```suggestion
    const inputData = this._tNode.inputs;
    let dataValue: PropertyAliasValue|undefined;
    if (inputData !== null && (dataValue = inputData[name])) {
      const lView = this._rootLView;
      setInputsForProperty(lView[TVIEW], lView, dataValue, name, value);
      markDirtyIfOnPush(lView, this._tNode.index);
    }
```

Reply:
Great suggestions, thnx @AndrewKushnir . I've added error "infrastructure", analogous to the one we've got for input bindings in a template.